### PR TITLE
Add -DHEADLESS option to cmake (nulled gfx)

### PR DESCRIPTION
**Pull request by ChillerDragon**
This allows compiling the client without graphics backend and thus it
can run also in a non graphical enviornment

## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [x] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -106,6 +106,7 @@ option(INFORM_UPDATE "Inform about available updates" ON)
 option(VIDEORECORDER "Enable video recording support via FFmpeg" OFF)
 option(UPNP "Enable UPnP support" OFF)
 option(ANTIBOT "Enable support for a dynamic anticheat library" OFF)
+option(HEADLESS "Build without graphics" OFF)
```
Comment (def-) [inline CMakeLists.txt:109]: HEADLESS_CLIENT maybe

Comment (ChillerDragon) [reply CMakeLists.txt:109]: Yea also thought about it but since server has no gfx anways i wanted to save keystrokes :D but sure ill update it

**Comment by Learath2** [comment]
Now this is what I call a cute patch. Thanks for reworking this into a nulled backend

**Comment by def-** [comment]
Doesn't hurt to have this I guess.
bors r+

**Pull request merged**